### PR TITLE
warn when creating PyOpenCLArrayContext without allocator

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: meshmode/array_context.py [hunk 1]
```diff
@@ -250,7 +250,13 @@ def __init__(self, queue, allocator=None):
         super().__init__()
         self.context = queue.context
         self.queue = queue
-        self.allocator = allocator
+        self.allocator = allocator if allocator else None
```
Comment (inducer) [inline meshmode/array_context.py:253]: I'm not sure `allocator=False` actually disables the warning.

File: meshmode/array_context.py [hunk 2]
```diff
@@ -250,7 +250,13 @@ def __init__(self, queue, allocator=None):
...
+        if allocator is None:
+            from warnings import warn
+            warn("PyOpenCLArrayContext created without an allocator. "
+                 "This can lead to high numbers of memory allocations. "
+                 "Please consider using a pyopencl.tools.MemoryPool. "
+                 "Run with allocator=False to disable this warning.")
```
Comment (inducer) [inline meshmode/array_context.py:259]: Also needs a doc update to describe the `allocator=False` argument.

File: meshmode/array_context.py (hunk 1 above)
Comment (matthiasdiener) [reply meshmode/array_context.py:253]: I just tested it again and it did disable the warning. Why would it not?

File: meshmode/array_context.py (hunk 2 above)
Comment (matthiasdiener) [reply meshmode/array_context.py:264]: I put a draft doc there.

File: meshmode/array_context.py [hunk 3]
```diff
@@ -244,13 +244,24 @@ class PyOpenCLArrayContext(ArrayContext):
         A :class:`pyopencl.CommandQueue`.
 
     .. attribute:: allocator
+        A PyOpenCL memory allocator. Can also be `None` (default) or `False` to
+        use the default allocator. Please note that running with the default
+        allocator will cause large numbers of memory allocations, and will
+        print a warning at runtime. Specify `False` to avoid the warning.
+        Please consider using a pyopencl.tools.MemoryPool.
```
Comment (inducer) [inline meshmode/array_context.py:251]:
```suggestion
        A PyOpenCL memory allocator. Can also be `None` (default) or `False` to
        use the default allocator. Please note that running with the default
        allocator allocates and deallocates OpenCL buffers directly. If lots
        of arrays are created (e.g. as results of computation), the associated cost
        may become significant. Using e.g. :class:`pyopencl.tools.MemoryPool`
        as the allocator can help avoid this cost.
```

**Comment by inducer** [comment]
Thanks!

**Pull request merged**